dialogue-client-test-lib/src/main/java/com/palantir/dialogue/AbstractChannelTest.java

```diff
@@ -67,13 +65,8 @@
 
     private final RequestBody body = new RequestBody() {
         @Override
-        public OptionalLong length() {
```

Comment:
when we know the length we probably want to write the Content-Length header

Reply:
Clients which buffer requests prior to sending them do send a content-length. This change allowed us to avoid fully buffering requests in many cases as we write directly to a socket.
HttpChannel (java 9 client) could set a content-length header, though it never did before.